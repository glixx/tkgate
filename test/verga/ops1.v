module top;

  reg [127:0] x,a,b,c;
  reg[31:0] y;

  initial
    begin
	a = 1652763623;
	b = 1287368762;
	c = 2763876432;
	x = a*b*c;
	$display("%d*%d*%d = %d",a,b,c,x);

	x = 32'hffffffff * 32'hffffffff;
	$display("x=%d x=%h",x,x);
	x = 32'hffffffff + 1;
	$display("x=%d x=%h",x,x);

	$display("%d * %d / %d = %d",a,b,c,a*b/c);

	a = 625376212786237651312;
	b = 127834612723233768273;
	$display("%d / %d = %d",a,b,a/b);
	$display("%d %% %d = %d",a,b,a%b);

	a = 625376212786237651312;
	b = 728176123827834612723233768273;
	$display("%d / %d = %d",a,b,a/b);
	$display("%d %% %d = %d",a,b,a%b);


	$display("%d >> 2 = %d",a,a>>2);
	$display("%d << 2 = %d",a,a<<2);
	$display("%d << 33 = %d",a,a<<33);
    end

endmodule
