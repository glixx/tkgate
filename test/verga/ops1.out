1652763623*1287368762*2763876432 = 5880744822841760707820497632
x=18446462594437873665 x=fffeffff00000001
x=4294967296 x=100000000
1652763623 * 1287368762 / 2763876432 = 769830457
625376212786237651312 / 127834612723233768273 = 4
625376212786237651312 % 127834612723233768273 = 114037761893302578220
625376212786237651312 / 728176123827834612723233768273 = 0
625376212786237651312 % 728176123827834612723233768273 = 625376212786237651312
625376212786237651312 >> 2 = 156344053196559412828
625376212786237651312 << 2 = 2501504851144950605248
625376212786237651312 << 33 = 5371940763226455502537782984704
